main/src/core/MonoDevelop.Core/MonoDevelop.Projects.FileNesting/FileNestingService.cs

```diff
@@ -52,16 +55,137 @@ static void HandleRulesProviderExtension (object sender, ExtensionNodeEventArgs
 			}
 		}
 
-		public static FilePath GetParentFile (Project project, FilePath inputFile)
+		public static bool HasParent (ProjectFile inputFile) => GetParentFile (inputFile) != null;
+
+		static ProjectNestingInfo GetProjectNestingInfo (Project project)
+		{
+			if (!loadedProjects.TryGetValue (project, out var nestingInfo)) {
+				nestingInfo = loadedProjects [project] = new ProjectNestingInfo (project);
+			}
+
+			return nestingInfo;
+		}
+
+		internal static ProjectFile InternalGetParentFile (ProjectFile inputFile)
 		{
+			if (inputFile.Project.UserProperties.HasValue ("FileNesting") && !inputFile.Project.UserProperties.GetValue<bool> ("FileNesting")) {
```

Comment:
Any reason we check for the value existing first?

```suggestion
			if (!inputFile.Project.UserProperties.GetValue<bool> ("FileNesting", false)) {
```

Reply:
Yeah, good one, fixed now, with a default of `true` instead, as that's why I was calling HasValue + GetValue, didn't see the overloaded method with the default